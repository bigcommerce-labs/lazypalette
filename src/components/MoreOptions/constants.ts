export const RestoreOriginalText = {
    action: 'Restore',
    body: 'All changes to this theme will be restored back to its default settings. ' +
    'This will not publish changes to your live store.',
    title: 'Restore original theme style',
};
