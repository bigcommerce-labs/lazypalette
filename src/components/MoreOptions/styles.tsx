import styled from 'styled-components';

export const ModalParagraph = styled.div`
    color: ${({ theme }) => theme.colors.primaryText};
    line-height: 1.5rem;
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    margin: 0 0 1.5rem 0;

    &:last-child {
        margin: 0;
    }
`;

ModalParagraph.defaultProps = {
    theme: {
        colors: {
            primaryText: '#FFFFFF',
        },
        typography: {
            fontSize: {
                small: '12px',
            },
        },
    },
};

export const List = styled.ul`
    cursor: pointer;
    width: 20rem;
    list-style: none;
    padding: 0;
    margin: 0 1rem 0 1.5rem;
`;

export const Item = styled.li`
    color: ${({theme}) => theme.colors.primaryText};
    line-height: 1.5rem;
    margin: 0 0 .5rem 0;

    :hover {
      text-decoration: underline
    }
`;

Item.defaultProps = {
    theme: {
        colors: {
            primaryText: '#FFFFFF',
        },
    },
};

export const Link = styled.a`
    color: ${({ theme }) => theme.colors.primary};
    text-decoration: none;
`;

Link.defaultProps = {
    theme: {
        colors: {
            primary: '#FFFFFF',
        },
    },
};

export const ExternalLink = styled.div`
    display: inline-block;
    margin: 0 0 0 0.25rem;
    position: relative;
    top: 0.125rem;
`;
