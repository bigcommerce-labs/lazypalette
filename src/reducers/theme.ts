import { CurrentThemeResponse, ThemeAction, ThemeActionTypes } from '../actions/theme';

export interface ThemeState {
  configurationId: string;
  schema: ThemeSchema;
  storeHash: string;
  themeVariations: ThemeVariations;
  versionId: string;
}

export interface ThemeSchema extends Array<ThemeSchemaEntry> {}

export interface ThemeSchemaEntry {
  name: string;
  settings: Array<{
    content?: string;
    force_reload?: boolean;
    id?: string;
    label?: string;
    options?: Array<{
      group?: string;
      label: string;
      value: string|number;
    }>;
    type: string;
  }>;
}

export interface ThemeVariations extends Array<ThemeVariationsEntry> {}

export interface ThemeVariationsEntry {
  configurationId: string;
  defaultConfigurationId: string;
  id: string;
  isCurrent: boolean;
  screenshot: {
    largePreview: string;
    largeThumb: string;
    smallThumb: string;
  };
  themeId: string;
  variationName: string;
}

const initialState: ThemeState = {
  configurationId: '',
  schema: [],
  storeHash: '',
  themeVariations: [],
  versionId: '',
};

const updateState = (oldState: ThemeState, newState: object) => ({ ...oldState, ...newState });

const setThemeResp = (state: ThemeState, data: CurrentThemeResponse) => {
  return updateState(state, {
    configurationId: data.configurationId,
    themeVariations: data.relatedVariations,
    versionId: data.versionId,
  });
};

function theme(state: ThemeState = initialState, action: ThemeAction): ThemeState {
  switch (action.type) {
    case ThemeActionTypes.CURRENT_THEME_RESPONSE:
      return setThemeResp(state, action.data);
    case ThemeActionTypes.THEME_CONFIG_RESPONSE:
      return updateState(state, { storeHash: action.data.storeHash });
    case ThemeActionTypes.THEME_VERSION_RESPONSE:
      return updateState(state, { schema: action.data.editorSchema });
    default:
      return state;
  }
}

export default theme;
