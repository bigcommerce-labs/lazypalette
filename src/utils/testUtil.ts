export const sel = (id: string) => `[data-test-id="${id}"]`;
