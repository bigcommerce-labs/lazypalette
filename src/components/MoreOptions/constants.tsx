import React from 'react';

import { Link, ModalParagraph } from './styles';

export const CopyThemeText = {
    action: 'Go to My Themes',
    body: `You must make a copy of your theme to edit the theme files.`,
    title: 'Edit Theme Files',
};

const designPolicyUrl = 'https://support.bigcommerce.com/articles/Public/BigCommerce-Design-Policy';

export const EditThemeFilesText = {
    action: 'Edit Theme Files',
    bodyActive:
        <>
            <ModalParagraph>
                It looks like you're about to modify your active theme by editing its theme files. Any changes you apply
                to your active theme will be made to your live store.
            </ModalParagraph>
            <ModalParagraph>
                Please note that BigCommerce is unable to offer support for any theme that has been modified with Edit
                Theme Files.&nbsp;
                <Link href={designPolicyUrl} target="_blank">
                    Learn more
                </Link>
            </ModalParagraph>
        </>,
    bodyInactive:
        <ModalParagraph>
            It looks like you're about to modify your theme by editing its theme files. Please note that BigCommerce
            is unable to offer support for any theme that has been modified with Edit Theme Files.&nbsp;
            <Link href={designPolicyUrl} target="_blank">
                Learn more
            </Link>
        </ModalParagraph>,
    title: 'Edit Theme Files',
};

export const RestoreOriginalText = {
    action: 'Restore',
    body: `All changes to this theme will be restored back to its default settings. This will not publish changes to
        your live store.`,
    title: 'Restore original theme style',
};

export enum CurrentModal {
    COPY_THEME = 'COPY_THEME',
    EDIT_THEME_FILES = 'EDIT_THEME_FILES',
    NONE = 'NONE',
    RESET = 'RESET',
}

export const FileEditorPath = (versionId: string, variationId: string, configurationId: string) =>
    `/manage/file-editor/${versionId}/${variationId}/${configurationId}`;

export const MyThemesPath = '/manage/storefront-manager/my-themes';
