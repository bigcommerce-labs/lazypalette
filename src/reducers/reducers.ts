import { combineReducers } from 'redux';

import error, { ErrorState } from './error';
import merchant, { MerchantStoreState } from './merchant';
import previewPane, { PreviewPaneState } from './previewPane';
import session, { SessionState } from './session';
import theme, { ThemeState } from './theme';

export interface State {
    merchant: MerchantStoreState;
    previewPane: PreviewPaneState;
    theme: ThemeState;
    error: ErrorState;
    session: SessionState;
}

const reducers = combineReducers({
    error,
    merchant,
    previewPane,
    session,
    theme,
});

export default reducers;
