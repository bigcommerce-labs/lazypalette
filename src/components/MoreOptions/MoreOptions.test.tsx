import { shallow } from 'enzyme';
import React from 'react';

import {ThemeVariationsEntry} from '../../reducers/theme';
import {sel} from '../../utils/testUtil';

import { MoreOptions } from './MoreOptions';

const mockProp: any = jest.fn();
const routeProps = {
    history: mockProp,
    location: mockProp,
    match: mockProp,
    staticContext: mockProp,
};
const loadTheme: any = jest.fn();

const variationEntry: ThemeVariationsEntry = {
    configurationId: '123',
    defaultConfigurationId: '234',
    id: '567',
    isCurrent: true,
    screenshot: {
        largePreview: 'host://meows/123.jpg',
        largeThumb: 'host://meows/234.jpg',
        smallThumb: 'host://meows/345.jpg',
    },
    themeId: '8900',
    variationName: 'light',
};

const restoreOriginalLink = sel('restore-original');

describe('<MoreOptions />', () => {
    const moreOptions = shallow(
        <MoreOptions
            position={{ x: 10, y: 10 }}
            variationId="234"
            currentVariationEntry={variationEntry}
            isChanged={false}
            loadTheme={loadTheme}
            {...routeProps}
        />
    );
    describe('render()', () => {
        it('renders the component', () => {
            expect(moreOptions).toMatchSnapshot();
        });
    });

    describe('when you click "Restore original theme styles"', () => {
        describe('when isChanged is true', () => {
            it('isResetOpen state should be true', () => {
                moreOptions.setProps({isChanged: true});
                moreOptions.find(restoreOriginalLink).simulate('click');
                expect(moreOptions.state().isResetOpen).toBe(true);
            });
        });

        describe('when isChanged is false', () => {
            it('should call loadTheme', () => {
                moreOptions.setProps({isChanged: false});
                moreOptions.find(restoreOriginalLink).simulate('click');
                expect(loadTheme).toHaveBeenCalled();
            });
        });
    });
});
