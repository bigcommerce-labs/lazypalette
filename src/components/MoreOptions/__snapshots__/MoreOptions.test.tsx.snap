// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoreOptions /> when you click "Edit Theme Files" when the design policy has not been acknowledged when viewing an inactive theme renders correctly 1`] = `
<MoreOptions
  activeThemeId="8900"
  canOptOut={true}
  configurationId="123"
  createNotification={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
          "Original theme styles restored.",
          "Success",
        ],
        Array [
          true,
          "Original theme styles restored.",
          "Success",
        ],
      ],
    }
  }
  currentVariationEntry={
    Object {
      "configurationId": "123",
      "defaultConfigurationId": "234",
      "id": "567",
      "isCurrent": true,
      "screenshot": Object {
        "largePreview": "host://meows/123.jpg",
        "largeThumb": "host://meows/234.jpg",
        "smallThumb": "host://meows/345.jpg",
      },
      "themeId": "8900",
      "variationName": "light",
    }
  }
  history={[MockFunction]}
  isChanged={false}
  isPrivate={true}
  loadTheme={
    [MockFunction] {
      "calls": Array [
        Array [
          "234",
          "234",
        ],
        Array [
          "234",
          "234",
        ],
      ],
    }
  }
  location={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "params": "",
      "path": "/",
      "url": "/",
    }
  }
  notifications={
    Object {
      "autoDismiss": true,
      "message": "hello",
      "type": "Success",
    }
  }
  position={
    Object {
      "x": 10,
      "y": 10,
    }
  }
  staticContext={[MockFunction]}
  storeHash="abc"
  themeId="8900"
  variationId="234"
  versionId="456"
>
  <Draggable
    position={
      Object {
        "x": 10,
        "y": 10,
      }
    }
  >
    <styled.div
      dragging={false}
      theme={
        Object {
          "layers": Object {
            "high": 100,
          },
        }
      }
    >
      <div
        className="sc-ckVGcZ cRvvSG"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
      >
        <ExpandableMenu
          back={
            Object {
              "pathname": "/",
              "search": undefined,
            }
          }
          position={
            Object {
              "x": 10,
              "y": 10,
            }
          }
          startMove={[Function]}
          startResize={[Function]}
          title="More Options"
          windowRef={
            Object {
              "current": <div
                class="sc-jKJlTe eNHUZn"
                style="left: 10px; max-height: 31rem; min-height: 10.5rem; top: 10px;"
              >
                <div
                  class="sc-eNQAEJ gNZsNX"
                >
                  <div
                    class="sc-kEYyzF jxrfWL"
                  >
                    <h2
                      class="sc-kkGfuU klAQHe"
                    >
                      More Options
                    </h2>
                    <a
                      class="sc-hMqMXs hOzlTA"
                      href="/"
                      theme="[object Object]"
                    />
                  </div>
                  <div
                    class="sc-iAyFgw gLJSFE"
                  >
                    <ul
                      class="sc-kgoBCf lnAFix"
                    >
                      <li
                        class="sc-kGXeez cMNWCR"
                        data-test-id="switch-to-theme-editor"
                      >
                        Switch to old Theme Editor
                      </li>
                      <li
                        class="sc-kGXeez cMNWCR"
                        data-test-id="edit-theme-files"
                      >
                        Edit Theme Files
                      </li>
                      <li
                        class="sc-kGXeez cMNWCR"
                        data-test-id="restore-original"
                      >
                        Restore original theme styles
                      </li>
                    </ul>
                  </div>
                  <div
                    class="sc-hSdWYo eXSuhC"
                  >
                    <div
                      class="sc-eHgmQL fbxFaP"
                    >
                      <span
                        class="sc-htpNat gUgXl"
                      >
                        <svg
                          class="sc-bxivhb fJCLdB"
                          version="1.1"
                          viewBox="0 0 16 6"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="
            M16 0H0V2H16V0ZM0 6H16V4H0V6Z
        "
                            fill="#CFD6E5"
                            stroke="none"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>,
            }
          }
        >
          <styled.div
            innerRef={
              Object {
                "current": <div
                  class="sc-jKJlTe eNHUZn"
                  style="left: 10px; max-height: 31rem; min-height: 10.5rem; top: 10px;"
                >
                  <div
                    class="sc-eNQAEJ gNZsNX"
                  >
                    <div
                      class="sc-kEYyzF jxrfWL"
                    >
                      <h2
                        class="sc-kkGfuU klAQHe"
                      >
                        More Options
                      </h2>
                      <a
                        class="sc-hMqMXs hOzlTA"
                        href="/"
                        theme="[object Object]"
                      />
                    </div>
                    <div
                      class="sc-iAyFgw gLJSFE"
                    >
                      <ul
                        class="sc-kgoBCf lnAFix"
                      >
                        <li
                          class="sc-kGXeez cMNWCR"
                          data-test-id="switch-to-theme-editor"
                        >
                          Switch to old Theme Editor
                        </li>
                        <li
                          class="sc-kGXeez cMNWCR"
                          data-test-id="edit-theme-files"
                        >
                          Edit Theme Files
                        </li>
                        <li
                          class="sc-kGXeez cMNWCR"
                          data-test-id="restore-original"
                        >
                          Restore original theme styles
                        </li>
                      </ul>
                    </div>
                    <div
                      class="sc-hSdWYo eXSuhC"
                    >
                      <div
                        class="sc-eHgmQL fbxFaP"
                      >
                        <span
                          class="sc-htpNat gUgXl"
                        >
                          <svg
                            class="sc-bxivhb fJCLdB"
                            version="1.1"
                            viewBox="0 0 16 6"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="
            M16 0H0V2H16V0ZM0 6H16V4H0V6Z
        "
                              fill="#CFD6E5"
                              stroke="none"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>,
              }
            }
            position={
              Object {
                "x": 10,
                "y": 10,
              }
            }
            theme={
              Object {
                "elevation": Object {
                  "floating": "0 2px 12px rgba(48, 53, 64, 0.2)",
                },
                "layers": Object {
                  "high": 50,
                },
              }
            }
          >
            <div
              className="sc-jKJlTe eNHUZn"
              style={
                Object {
                  "left": "10px",
                  "maxHeight": "31rem",
                  "minHeight": "10.5rem",
                  "top": "10px",
                }
              }
            >
              <styled.div>
                <div
                  className="sc-eNQAEJ gNZsNX"
                >
                  <styled.div
                    theme={
                      Object {
                        "colors": Object {
                          "background": "#F5F7FA",
                          "primaryText": "#AAAAAA",
                        },
                        "typography": Object {
                          "fontWeight": Object {
                            "normal": 400,
                          },
                        },
                      }
                    }
                  >
                    <div
                      className="sc-kEYyzF jxrfWL"
                    >
                      <styled.h2
                        onMouseDown={[Function]}
                        theme={
                          Object {
                            "colors": Object {
                              "primaryText": "#AAAAAA",
                            },
                            "typography": Object {
                              "fontWeight": Object {
                                "normal": 400,
                              },
                            },
                          }
                        }
                      >
                        <h2
                          className="sc-kkGfuU klAQHe"
                          onMouseDown={[Function]}
                        >
                          More Options
                        </h2>
                      </styled.h2>
                      <Styled(Link)
                        onClick={[Function]}
                        replace={true}
                        theme={
                          Object {
                            "colors": Object {
                              "background": "#000000",
                              "primaryText": "#AAAAAA",
                            },
                          }
                        }
                        to={
                          Object {
                            "pathname": "/",
                            "search": undefined,
                          }
                        }
                      >
                        <Link
                          className="sc-hMqMXs hOzlTA"
                          onClick={[Function]}
                          replace={true}
                          theme={
                            Object {
                              "colors": Object {
                                "background": "#000000",
                                "primaryText": "#AAAAAA",
                              },
                            }
                          }
                          to={
                            Object {
                              "pathname": "/",
                              "search": undefined,
                            }
                          }
                        >
                          <a
                            className="sc-hMqMXs hOzlTA"
                            href="/"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "background": "#000000",
                                  "primaryText": "#AAAAAA",
                                },
                              }
                            }
                          />
                        </Link>
                      </Styled(Link)>
                    </div>
                  </styled.div>
                  <styled.div
                    theme={
                      Object {
                        "colors": Object {
                          "empty": "#FFFFFF",
                        },
                      }
                    }
                  >
                    <div
                      className="sc-iAyFgw gLJSFE"
                    >
                      <styled.ul>
                        <ul
                          className="sc-kgoBCf lnAFix"
                        >
                          <styled.li
                            data-test-id="switch-to-theme-editor"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "primaryText": "#FFFFFF",
                                },
                              }
                            }
                          >
                            <li
                              className="sc-kGXeez cMNWCR"
                              data-test-id="switch-to-theme-editor"
                              onClick={[Function]}
                            >
                              Switch to old Theme Editor
                            </li>
                          </styled.li>
                          <styled.li
                            data-test-id="edit-theme-files"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "primaryText": "#FFFFFF",
                                },
                              }
                            }
                          >
                            <li
                              className="sc-kGXeez cMNWCR"
                              data-test-id="edit-theme-files"
                              onClick={[Function]}
                            >
                              Edit Theme Files
                            </li>
                          </styled.li>
                          <styled.li
                            data-test-id="restore-original"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "primaryText": "#FFFFFF",
                                },
                              }
                            }
                          >
                            <li
                              className="sc-kGXeez cMNWCR"
                              data-test-id="restore-original"
                              onClick={[Function]}
                            >
                              Restore original theme styles
                            </li>
                          </styled.li>
                        </ul>
                      </styled.ul>
                    </div>
                  </styled.div>
                  <styled.div
                    onMouseDown={[Function]}
                  >
                    <div
                      className="sc-hSdWYo eXSuhC"
                      onMouseDown={[Function]}
                    >
                      <styled.div>
                        <div
                          className="sc-eHgmQL fbxFaP"
                        >
                          <Icon
                            glyph="resizeable"
                            primaryColor="#CFD6E5"
                            size="large"
                          >
                            <styled.span
                              iconSize="large"
                            >
                              <span
                                className="sc-htpNat gUgXl"
                              >
                                <styled.svg
                                  viewBox="0 0 16 6"
                                >
                                  <svg
                                    className="sc-bxivhb fJCLdB"
                                    version="1.1"
                                    viewBox="0 0 16 6"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="
            M16 0H0V2H16V0ZM0 6H16V4H0V6Z
        "
                                      fill="#CFD6E5"
                                      stroke="none"
                                    />
                                  </svg>
                                </styled.svg>
                              </span>
                            </styled.span>
                          </Icon>
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </ExpandableMenu>
      </div>
    </styled.div>
  </Draggable>
</MoreOptions>
`;

exports[`<MoreOptions /> when you click "Edit Theme Files" when the design policy has not been acknowledged when viewing the active theme renders correctly 1`] = `
<MoreOptions
  activeThemeId="8900"
  canOptOut={true}
  configurationId="123"
  createNotification={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
          "Original theme styles restored.",
          "Success",
        ],
        Array [
          true,
          "Original theme styles restored.",
          "Success",
        ],
      ],
    }
  }
  currentVariationEntry={
    Object {
      "configurationId": "123",
      "defaultConfigurationId": "234",
      "id": "567",
      "isCurrent": true,
      "screenshot": Object {
        "largePreview": "host://meows/123.jpg",
        "largeThumb": "host://meows/234.jpg",
        "smallThumb": "host://meows/345.jpg",
      },
      "themeId": "8900",
      "variationName": "light",
    }
  }
  history={[MockFunction]}
  isChanged={false}
  isPrivate={true}
  loadTheme={
    [MockFunction] {
      "calls": Array [
        Array [
          "234",
          "234",
        ],
        Array [
          "234",
          "234",
        ],
      ],
    }
  }
  location={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "params": "",
      "path": "/",
      "url": "/",
    }
  }
  notifications={
    Object {
      "autoDismiss": true,
      "message": "hello",
      "type": "Success",
    }
  }
  position={
    Object {
      "x": 10,
      "y": 10,
    }
  }
  staticContext={[MockFunction]}
  storeHash="abc"
  themeId="8900"
  variationId="234"
  versionId="456"
>
  <Draggable
    position={
      Object {
        "x": 10,
        "y": 10,
      }
    }
  >
    <styled.div
      dragging={false}
      theme={
        Object {
          "layers": Object {
            "high": 100,
          },
        }
      }
    >
      <div
        className="sc-ckVGcZ cRvvSG"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
      >
        <ExpandableMenu
          back={
            Object {
              "pathname": "/",
              "search": undefined,
            }
          }
          position={
            Object {
              "x": 10,
              "y": 10,
            }
          }
          startMove={[Function]}
          startResize={[Function]}
          title="More Options"
          windowRef={
            Object {
              "current": <div
                class="sc-jKJlTe eNHUZn"
                style="left: 10px; max-height: 31rem; min-height: 10.5rem; top: 10px;"
              >
                <div
                  class="sc-eNQAEJ gNZsNX"
                >
                  <div
                    class="sc-kEYyzF jxrfWL"
                  >
                    <h2
                      class="sc-kkGfuU klAQHe"
                    >
                      More Options
                    </h2>
                    <a
                      class="sc-hMqMXs hOzlTA"
                      href="/"
                      theme="[object Object]"
                    />
                  </div>
                  <div
                    class="sc-iAyFgw gLJSFE"
                  >
                    <ul
                      class="sc-kgoBCf lnAFix"
                    >
                      <li
                        class="sc-kGXeez cMNWCR"
                        data-test-id="switch-to-theme-editor"
                      >
                        Switch to old Theme Editor
                      </li>
                      <li
                        class="sc-kGXeez cMNWCR"
                        data-test-id="edit-theme-files"
                      >
                        Edit Theme Files
                      </li>
                      <li
                        class="sc-kGXeez cMNWCR"
                        data-test-id="restore-original"
                      >
                        Restore original theme styles
                      </li>
                    </ul>
                  </div>
                  <div
                    class="sc-hSdWYo eXSuhC"
                  >
                    <div
                      class="sc-eHgmQL fbxFaP"
                    >
                      <span
                        class="sc-htpNat gUgXl"
                      >
                        <svg
                          class="sc-bxivhb fJCLdB"
                          version="1.1"
                          viewBox="0 0 16 6"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="
            M16 0H0V2H16V0ZM0 6H16V4H0V6Z
        "
                            fill="#CFD6E5"
                            stroke="none"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>,
            }
          }
        >
          <styled.div
            innerRef={
              Object {
                "current": <div
                  class="sc-jKJlTe eNHUZn"
                  style="left: 10px; max-height: 31rem; min-height: 10.5rem; top: 10px;"
                >
                  <div
                    class="sc-eNQAEJ gNZsNX"
                  >
                    <div
                      class="sc-kEYyzF jxrfWL"
                    >
                      <h2
                        class="sc-kkGfuU klAQHe"
                      >
                        More Options
                      </h2>
                      <a
                        class="sc-hMqMXs hOzlTA"
                        href="/"
                        theme="[object Object]"
                      />
                    </div>
                    <div
                      class="sc-iAyFgw gLJSFE"
                    >
                      <ul
                        class="sc-kgoBCf lnAFix"
                      >
                        <li
                          class="sc-kGXeez cMNWCR"
                          data-test-id="switch-to-theme-editor"
                        >
                          Switch to old Theme Editor
                        </li>
                        <li
                          class="sc-kGXeez cMNWCR"
                          data-test-id="edit-theme-files"
                        >
                          Edit Theme Files
                        </li>
                        <li
                          class="sc-kGXeez cMNWCR"
                          data-test-id="restore-original"
                        >
                          Restore original theme styles
                        </li>
                      </ul>
                    </div>
                    <div
                      class="sc-hSdWYo eXSuhC"
                    >
                      <div
                        class="sc-eHgmQL fbxFaP"
                      >
                        <span
                          class="sc-htpNat gUgXl"
                        >
                          <svg
                            class="sc-bxivhb fJCLdB"
                            version="1.1"
                            viewBox="0 0 16 6"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="
            M16 0H0V2H16V0ZM0 6H16V4H0V6Z
        "
                              fill="#CFD6E5"
                              stroke="none"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>,
              }
            }
            position={
              Object {
                "x": 10,
                "y": 10,
              }
            }
            theme={
              Object {
                "elevation": Object {
                  "floating": "0 2px 12px rgba(48, 53, 64, 0.2)",
                },
                "layers": Object {
                  "high": 50,
                },
              }
            }
          >
            <div
              className="sc-jKJlTe eNHUZn"
              style={
                Object {
                  "left": "10px",
                  "maxHeight": "31rem",
                  "minHeight": "10.5rem",
                  "top": "10px",
                }
              }
            >
              <styled.div>
                <div
                  className="sc-eNQAEJ gNZsNX"
                >
                  <styled.div
                    theme={
                      Object {
                        "colors": Object {
                          "background": "#F5F7FA",
                          "primaryText": "#AAAAAA",
                        },
                        "typography": Object {
                          "fontWeight": Object {
                            "normal": 400,
                          },
                        },
                      }
                    }
                  >
                    <div
                      className="sc-kEYyzF jxrfWL"
                    >
                      <styled.h2
                        onMouseDown={[Function]}
                        theme={
                          Object {
                            "colors": Object {
                              "primaryText": "#AAAAAA",
                            },
                            "typography": Object {
                              "fontWeight": Object {
                                "normal": 400,
                              },
                            },
                          }
                        }
                      >
                        <h2
                          className="sc-kkGfuU klAQHe"
                          onMouseDown={[Function]}
                        >
                          More Options
                        </h2>
                      </styled.h2>
                      <Styled(Link)
                        onClick={[Function]}
                        replace={true}
                        theme={
                          Object {
                            "colors": Object {
                              "background": "#000000",
                              "primaryText": "#AAAAAA",
                            },
                          }
                        }
                        to={
                          Object {
                            "pathname": "/",
                            "search": undefined,
                          }
                        }
                      >
                        <Link
                          className="sc-hMqMXs hOzlTA"
                          onClick={[Function]}
                          replace={true}
                          theme={
                            Object {
                              "colors": Object {
                                "background": "#000000",
                                "primaryText": "#AAAAAA",
                              },
                            }
                          }
                          to={
                            Object {
                              "pathname": "/",
                              "search": undefined,
                            }
                          }
                        >
                          <a
                            className="sc-hMqMXs hOzlTA"
                            href="/"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "background": "#000000",
                                  "primaryText": "#AAAAAA",
                                },
                              }
                            }
                          />
                        </Link>
                      </Styled(Link)>
                    </div>
                  </styled.div>
                  <styled.div
                    theme={
                      Object {
                        "colors": Object {
                          "empty": "#FFFFFF",
                        },
                      }
                    }
                  >
                    <div
                      className="sc-iAyFgw gLJSFE"
                    >
                      <styled.ul>
                        <ul
                          className="sc-kgoBCf lnAFix"
                        >
                          <styled.li
                            data-test-id="switch-to-theme-editor"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "primaryText": "#FFFFFF",
                                },
                              }
                            }
                          >
                            <li
                              className="sc-kGXeez cMNWCR"
                              data-test-id="switch-to-theme-editor"
                              onClick={[Function]}
                            >
                              Switch to old Theme Editor
                            </li>
                          </styled.li>
                          <styled.li
                            data-test-id="edit-theme-files"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "primaryText": "#FFFFFF",
                                },
                              }
                            }
                          >
                            <li
                              className="sc-kGXeez cMNWCR"
                              data-test-id="edit-theme-files"
                              onClick={[Function]}
                            >
                              Edit Theme Files
                            </li>
                          </styled.li>
                          <styled.li
                            data-test-id="restore-original"
                            onClick={[Function]}
                            theme={
                              Object {
                                "colors": Object {
                                  "primaryText": "#FFFFFF",
                                },
                              }
                            }
                          >
                            <li
                              className="sc-kGXeez cMNWCR"
                              data-test-id="restore-original"
                              onClick={[Function]}
                            >
                              Restore original theme styles
                            </li>
                          </styled.li>
                        </ul>
                      </styled.ul>
                    </div>
                  </styled.div>
                  <styled.div
                    onMouseDown={[Function]}
                  >
                    <div
                      className="sc-hSdWYo eXSuhC"
                      onMouseDown={[Function]}
                    >
                      <styled.div>
                        <div
                          className="sc-eHgmQL fbxFaP"
                        >
                          <Icon
                            glyph="resizeable"
                            primaryColor="#CFD6E5"
                            size="large"
                          >
                            <styled.span
                              iconSize="large"
                            >
                              <span
                                className="sc-htpNat gUgXl"
                              >
                                <styled.svg
                                  viewBox="0 0 16 6"
                                >
                                  <svg
                                    className="sc-bxivhb fJCLdB"
                                    version="1.1"
                                    viewBox="0 0 16 6"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="
            M16 0H0V2H16V0ZM0 6H16V4H0V6Z
        "
                                      fill="#CFD6E5"
                                      stroke="none"
                                    />
                                  </svg>
                                </styled.svg>
                              </span>
                            </styled.span>
                          </Icon>
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </ExpandableMenu>
      </div>
    </styled.div>
  </Draggable>
</MoreOptions>
`;
